src/pybind/mgr/tests/__init__.py

```diff
@@ -17,18 +17,38 @@
         import mock
 
     class M(object):
+        def _ceph_get_store(self, k):
+            return self._store.get(k, None)
+
+        def _ceph_set_store(self, k, v):
+            if v is None:
+                if k in self._store:
+                    del self._store[k]
+            else:
+                self._store[k] = v
+
+        def _ceph_get_store_prefix(self, prefix):
+            return {
+                k: v for k, v in self._store.items()
+                if k.startswith(prefix)
+            }
+
+        def _ceph_get_module_option(self, module, key, localized_prefix: None):
+            return self.get_store(f'{module}/{key}')
+
+        def _ceph_set_module_option(self, module, key, val):
+            return self.set_store(f'{module}/{key}', val)
```

Comment:
Maybe it's just me that I'm a bit thick today, but where does this method come from?

Reply:
https://github.com/ceph/ceph/blob/9b40f6714ff0d8ce2ac9dffb27e641d36253df7e/src/mgr/BaseMgrModule.cc#L1117-L1118

This is how `ceph_module` kind of looks like:

https://github.com/ceph/ceph/blob/36f5e3b115584df022186965dbcb29282376d091/src/pybind/mgr/ceph_module.pyi

Reply:
following the principle that you should not mock anything in your code base, which also includes `mgr_module.py`, but not the C++ interface, I mocked it there.

Reply:
Yeah, got that. What I don't still see is who's providing `set_store` implementation? That M class doesn't inherit from anything else...

Reply:
https://github.com/ceph/ceph/blob/90699f6e48c322ec8612939617e26df6305b8da1/src/pybind/mgr/mgr_module.py#L1314-L1322

Reply:
let me call `_ceph_set_store` directly. hold on

Reply:
fixed